application/backend/app/services/data_collect/data_collector.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from abc import ABCMeta, abstractmethod
+from uuid import UUID
+
+import numpy as np
+
+from app.entities.stream_data import InferenceData
+from app.schemas import Project
+from app.schemas.dataset_item import DatasetItemFormat
+from app.schemas.pipeline import FixedRateDataCollectionPolicy
+from app.services import ActivePipelineService, DatasetService
+from app.services.data_collect.prediction_converter import convert_prediction
+from app.settings import get_settings
+from app.utils import Singleton
+
+logger = logging.getLogger(__name__)
+
+
+class PolicyChecker(metaclass=ABCMeta):
+    @abstractmethod
+    def should_collect(self, timestamp: float) -> bool:
+        """
+        Checks if certain dispatched image should be collected to the project dataset
+        :param timestamp: dispatched image timestamp
```

Comment:
```suggestion
        Checks if certain dispatched image should be collected to the project dataset
        
        :param timestamp: dispatched image timestamp
```
Please leave a blank line between the description and the parameters, otherwise PyCharm and other editors don't render it correctly